google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -1430,11 +1438,78 @@ public String getNotFoundPage() {
    */
   @Deprecated
   public List<? extends DeleteRule> getDeleteRules() {
-    return deleteRules;
+    return deleteRules != null ? deleteRules : ImmutableList.<DeleteRule>of();
   }
 
   public List<? extends LifecycleRule> getLifecycleRules() {
-    return lifecycleRules;
+    return lifecycleRules != null ? lifecycleRules : ImmutableList.<LifecycleRule>of();
+  }
+
+  /**
+   * Deletes the lifecycle rules of this bucket.
+   *
+   * <p>Example of deleting the lifecycle rules of this bucket:
+   *
+   * <pre>{@code
+   * String bucketName = "my-unique-bucket";
+   * LifecycleRule lifecycleRule_1 =
+   * 	new LifecycleRule(
+   * 		LifecycleAction.newSetStorageClassAction(StorageClass.COLDLINE),
+   * 		LifecycleCondition.newBuilder()
+   * 			.setAge(1)
+   * 			.setNumberOfNewerVersions(3)
+   * 			.setIsLive(false)
+   * 			.setMatchesStorageClass(ImmutableList.of(StorageClass.COLDLINE))
+   * 			.build());
+   * LifecycleRule lifecycleRule_2 =
+   * 	new LifecycleRule(
+   * 		LifecycleAction.newDeleteAction(), LifecycleCondition.newBuilder().setAge(1).build());
+   * ImmutableList<LifecycleRule> lifecycleRules =
+   * 	ImmutableList.of(lifecycleRule_1, lifecycleRule_2);
+   * Bucket bucket =
+   * 	storage.create(
+   * 		BucketInfo.newBuilder(bucketName)
+   * 			.setLocation("us")
+   * 			.setLifecycleRules(lifecycleRules)
+   * 			.build());
+   * List<LifecycleRule> results = bucket.deleteLifecycleRules(storage, lifecycleRule_1);
+   * }</pre>
+   *
+   * @param rulesToDelete the set of lifecycle rules to delete
+   * @param storage an object of storage
+   * @return the lists of deleted lifecycle rules of bucket, an empty list if the requested
+   *     lifecycle rules was not found
+   * @throws StorageException upon failure
+   */
+  public List<LifecycleRule> deleteLifecycleRules(Storage storage, LifecycleRule... rulesToDelete) {
```

Comment:
@dmitry-fa 
There is no business logic inside the getLifecylceRules() where the storage object is needed.
`deleteLifecycleRules() `has business logic like here with the help of the storage object we take the actual rules of the bucket and then compare them with the passed rules If the two rules are the same then it will be removed from the bucket's `LifecycleRules `list So I don't think this would be possible without the object of `storage`.

let me know if you have better idea to get the actual rules of bucket without using `storage `object.

Reply:
It should be implemented in the same manner as other methods updating Bucket's properties:
```
Bucket updatedBucket = bucket
    .toBuilder()
    .setVersioningEnabled(true)
    .deleteLifecycleRules()
    .build()
    .update();
```
implemented like:
```
    @Override
    public Builder deleteLifecycleRules() {
      infoBuilder.deleteLifecycleRules();
      return this;
    }
```

Reply:
See also [setLifecycleRules](https://googleapis.dev/java/google-cloud-storage/latest/com/google/cloud/storage/Bucket.Builder.html#setLifecycleRules-java.lang.Iterable-)

Reply:
@dmitry-fa Thanks for the suggestion

We Cannot override such method because `deleteLifecycleRule `does not present inside `bucket proto` model. FYI `BucketInfo `contain only those properties which are present inside the `proto `model.